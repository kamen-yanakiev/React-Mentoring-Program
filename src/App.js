import React from 'react';
import Nav from './Nav';
import Sidebar from './Sidebar';
import Search from './Search';

function App() {
  return (
    <div className="App">
        <Nav/>
        <Search></Search>
        <Sidebar></Sidebar>
    </div>
  );
}

export default App;
