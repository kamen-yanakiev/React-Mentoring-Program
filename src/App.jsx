import React from 'react';
import {
    BrowserRouter as Router, Switch, Route, Redirect,
} from 'react-router-dom';
import MainContentHolder from './pages/MainContentHolder';
import SelectedMovieHeader from './components/header/SelectedMovieHeader';
import ErrorPage from './pages/ErrorPage';

const App = () => (
    <Router>
        <div className='App'>
            <Switch>
                <Route exact path='/'>
                    <Redirect to='/movies' />
                </Route>
                <Route exact path='/movies' component={MainContentHolder} />
                <Route exact path='/movie/:id' component={SelectedMovieHeader} />
                <Route path='/' component={ErrorPage} />
            </Switch>
        </div>
    </Router>
);

export default App;
