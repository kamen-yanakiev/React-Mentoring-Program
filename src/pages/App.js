import React from 'react';
import Header from './Header';
import Sidebar from './Sidebar';
import MainContent from './MainContent'

function App() {
  return (
    <div className="App">
        <Header/>
        <MainContent></MainContent>
        <Sidebar></Sidebar>
    </div>
  );
}

export default App;
