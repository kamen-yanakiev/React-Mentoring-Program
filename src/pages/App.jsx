import React, { useState, useEffect, useCallback } from "react";
import Header from "./Header";
import MainContent from "./MainContent";

const App = () => {
  const [loading, clickHandler] = useState(false);

  const [movieToShow, showMovieData] = useState({});

  const movieToOpen = (movie) => {
    memoizedCallback(movie.title, movie.id);
    showMovieData(movie);
  };

  const closeMovie = () => {
    clickHandler(false);
    showMovieData({});
  };

  const memoizedCallback = useCallback((name, id) => {
    console.log(`Inside useCallback -> ${name} --- ${id}`);
  }, []);

  useEffect(() => {
    if (movieToShow.title) {
      clickHandler(true);
    }
  }, [movieToShow]);

  return (
    <div className="App">
      <Header
        loading={loading}
        passMovieToClose={closeMovie}
        movie={movieToShow}
      />
      <p></p>
      <MainContent openMovie={movieToOpen} />
    </div>
  );
};

export default App;
