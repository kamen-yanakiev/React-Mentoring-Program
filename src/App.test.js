import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { createStore } from 'redux';
import reducer from './context/movies/reducers';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import renderer from 'react-test-renderer';
import App from "./App";

configure({ adapter: new Adapter() });
const store = createStore(reducer);
it('renders correctly', () => {
    const tree = renderer
        .create(<Provider store={store}>
            <App />
        </Provider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

describe("App", () => {
    test("Verify that it is created", () => {
        const component = mount(
            <Provider store={store}>
                <App />
            </Provider>);
        expect(component.find(App)).toHaveLength(1);
    });
});