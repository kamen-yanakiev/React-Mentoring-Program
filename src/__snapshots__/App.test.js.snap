// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="header"
    >
      <div
        className="header-text"
      >
        <span
          className="netflix-span"
        >
          netflix
        </span>
        roulette
      </div>
      <div
        className="edit-movie-btn"
      >
        <button
          className="add-movie-btn btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          + Add Movie
        </button>
      </div>
      <div
        className="search-and-results"
      >
        <h2
          className="find-movie"
        >
          Find your movie
        </h2>
        <input
          className="search-movie-input"
          id=""
          name="searchbar"
          placeholder="Search for your movie here"
          type="text"
        />
        <button
          className="movie-search-btn"
        >
          Search
        </button>
      </div>
    </div>
    <img
      alt="loading..."
      src="i-am-a-stubbed-file"
      style={
        Object {
          "display": "block",
          "margin": "auto",
          "width": "200px",
        }
      }
    />
  </div>
</div>
`;
